metacat-common-server/src/main/java/com/netflix/metacat/common/server/api/authorization/Authorization.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ *  Copyright 2024 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.metacat.common.server.api.authorization;
+
+import com.netflix.metacat.common.MetacatRequestContext;
+import lombok.NonNull;
+
+/**
+ * And interface for the authorization of requests based on request context.
+ *
+ * @author abozigian
+ */
+public interface Authorization {
+
+    /**
+     * Authorize a given request based on the request context.
+     *
+     * @param context The request context.
+     * @return True if the request is authorized and False if unauthorized.
+     */
+    default AuthorizationStatus isAuthorized(@NonNull MetacatRequestContext context) {
```

Comment:
The interface could be more explicit about its expectations from the MetacatRequestContext, instead of passing down the entire context.